File: sort-mapper-example-java/src/main/java/kh/farrukh/sortmapperexamplejava/teacher/TeacherRepo.java
package kh.farrukh.sortmapperexamplejava.teacher;

import java.util.Optional;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface TeacherRepo extends JpaRepository<TeacherEntity, Long> {

  Optional<TeacherEntity> findByIdAndDeletedAtIsNull(Long id);

  Page<TeacherEntity> findAllByDeletedAtIsNull(Pageable pageable);

}

File: sort-mapper-example/src/main/kotlin/kh/farrukh/sortmapperexample/teacher/TeacherRepo.kt
package kh.farrukh.sortmapperexample.teacher

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface TeacherRepo : JpaRepository<TeacherEntity, Long> {

    fun findByIdAndDeletedAtIsNull(id: Long): TeacherEntity?

    fun findAllByDeletedAtIsNull(pageable: Pageable): Page<TeacherEntity>

}